src/miral/CMakeLists.txt

```diff
@@ -29,6 +29,7 @@ add_library(miral-internal STATIC
                                          join_client_threads.h
                                          window_info_defaults.h
     window_specification_internal.cpp    window_specification_internal.h
+    application_selector.cpp
```

Comment:
This belongs in `miral`, not `miral-internal` - we need to publish the associated entry points